etl/parse_email_save_attachment/parse_messages.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+
+import os
+import pytz
+import boto3
+import email
+import tempfile
+from email import policy
+from operator import itemgetter
+from datetime import datetime
+
+
+def get_most_recent_file(bucket_name, prefix):
+    s3 = boto3.client("s3")
+
+    # List objects within the specified bucket and prefix
+    response = s3.list_objects_v2(Bucket=bucket_name, Prefix=prefix)
```

Comment:
yep, this is an annoying limitation we've run into elsewhere. what if you ran a cleanup function at the end of the script that moved the file to another subdirectory in the buck?

Reply:
This is a TIL for me! 🙏 Looks like there may be a [way to paginate](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/paginator/ListObjectsV2.html#)? Not really sure if this fits in here

Reply:
Great call y'all - I had not considered this gotcha. I'm going to address this in two ways:

1) Fix the code to paginate
2) Setup life cycle rules on the S3 bucket so that old data gets expired and removed. This will prevent us from building up more than 1000 items, and also just keep old, unneeded data in the bit bucket and not ever-increasing our AWS bill. 

Here's the life cycle summary I'm putting in place:

![image](https://github.com/cityofaustin/dts-maximo-geo-integration/assets/3653206/b89097d3-225e-4b8c-96cc-48bf10ed4c57)

Reply:
Pagination code added here: 00582aac07e79b9c9419ea92dc234663c9bbed91